pkg/telemetrymetadata/body_json_metadata_test.go

```diff
@@ -0,0 +1,209 @@
+	testCases := []struct {
+		name              string
+		fieldKeySelectors []*telemetrytypes.FieldKeySelector
+		expectedSQL       []string // SQL should contain these strings
+		notExpectedSQL    []string // SQL should NOT contain these strings
+		expectedArgs      []any    // Args include search texts (if any) + limit value
+		expectedLimit     int
+	}{
+
+		{
+			name: "Single search text with EQUAL operator",
+			fieldKeySelectors: []*telemetrytypes.FieldKeySelector{
+				{
+					Name:              "user.name",
+					SelectorMatchType: telemetrytypes.FieldSelectorMatchTypeExact,
+				},
+			},
+			expectedSQL: []string{
+				"SELECT",
+				"path",
+				"groupArray(DISTINCT type) AS types",
+				"max(last_seen) AS last_seen",
+				fmt.Sprintf("FROM %s.%s", DBName, PathTypesTableName),
+				"WHERE",
+				"path = ?",
+				"GROUP BY path",
+				"ORDER BY last_seen DESC",
+				"LIMIT ?",
+			},
+			notExpectedSQL: []string{
+				"ILIKE",
+				"LIKE",
+			},
+			expectedArgs:  []any{"user.name", defaultPathLimit},
+			expectedLimit: defaultPathLimit,
+		},
+		{
+			name: "Single search text with LIKE operator",
+			fieldKeySelectors: []*telemetrytypes.FieldKeySelector{
+				{
+					Name:              "user",
+					SelectorMatchType: telemetrytypes.FieldSelectorMatchTypeFuzzy,
+				},
+			},
+			expectedSQL: []string{
+				"SELECT",
+				"path",
+				"groupArray(DISTINCT type) AS types",
+				"max(last_seen) AS last_seen",
+				fmt.Sprintf("FROM %s.%s", DBName, PathTypesTableName),
+				"WHERE",
+				"path LIKE ?",
+				"GROUP BY path",
+				"ORDER BY last_seen DESC",
+				"LIMIT ?",
+			},
+			notExpectedSQL: []string{
+				"path = ?",
+			},
+			expectedArgs:  []any{"user", 100},
+			expectedLimit: 100,
+		},
+		{
+			name: "Multiple search texts with EQUAL operator",
+			fieldKeySelectors: []*telemetrytypes.FieldKeySelector{
+				{
+					Name:              "user.name",
+					SelectorMatchType: telemetrytypes.FieldSelectorMatchTypeExact,
+				},
+				{
+					Name:              "user.age",
+					SelectorMatchType: telemetrytypes.FieldSelectorMatchTypeExact,
+				},
+			},
+			expectedSQL: []string{
+				"SELECT",
+				"path",
+				"groupArray(DISTINCT type) AS types",
+				"max(last_seen) AS last_seen",
+				fmt.Sprintf("FROM %s.%s", DBName, PathTypesTableName),
+				"WHERE",
+				"OR",
+				"path = ?",
+				"GROUP BY path",
+				"ORDER BY last_seen DESC",
+				"LIMIT ?",
+			},
+			notExpectedSQL: []string{
+				"ILIKE",
+				"LIKE",
+			},
+			expectedArgs:  []any{"user.name", "user.age", defaultPathLimit},
+			expectedLimit: defaultPathLimit,
+		},
+		{
+			name: "Multiple search texts with LIKE operator",
+			fieldKeySelectors: []*telemetrytypes.FieldKeySelector{
+				{
+					Name:              "user",
+					SelectorMatchType: telemetrytypes.FieldSelectorMatchTypeFuzzy,
+				},
+				{
+					Name:              "admin",
+					SelectorMatchType: telemetrytypes.FieldSelectorMatchTypeFuzzy,
+				},
+			},
+			expectedSQL: []string{
+				"SELECT",
```

Comment:
we can make this better just by looking at this
```
SELECT path groupArray(DISTINCT type) AS types max(last_seen) AS last_seen 
FROM signoz_metadata.distributed_json_path_types 
WHERE OR path = ? GROUP BY path ORDER BY last_seen DESC LIMIT ?
```
but the actual value that should be compared is 
```
SELECT path, groupArray(DISTINCT type) AS types, max(last_seen) AS last_seen 
FROM signoz_metadata.distributed_json_path_types 
WHERE (path = ? OR path = ?) GROUP BY path ORDER BY last_seen DESC LIMIT ?
```

now since your main paint here is to compare if the `Multiple search texts with LIKE operator` the test doesn't fully check it.